<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>597</width>
    <height>503</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FAB-tweak-tom</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="introPage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Please select a tweak method from the menu above...</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="bedLevelGroupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Milled Bed Level G-Code Creator</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="7" column="0">
              <widget class="QLabel" name="bedLevelSpindleSpeedLabel">
               <property name="text">
                <string>Spindle Speed (RPM) :</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="overlapSizeLabel">
               <property name="text">
                <string>Overlap (in mm) :</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelOverlapSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="maximum">
                <double>214.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>0.150000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelMillSizeSpinbox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="maximum">
                <double>214.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>0.300000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="widthLabel">
               <property name="text">
                <string>Width (in mm) :</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelHeightSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="maximum">
                <double>236.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelWidthSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="maximum">
                <double>214.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="heightLabel">
               <property name="text">
                <string>Height (in mm) :</string>
               </property>
              </widget>
             </item>
             <item row="3" column="2">
              <spacer name="horizontalSpacer_5">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="millSizeLabel">
               <property name="toolTip">
                <string>The size of the mill bit at the depth you want leveled.</string>
               </property>
               <property name="text">
                <string>Mill size (in mm) :</string>
               </property>
              </widget>
             </item>
             <item row="7" column="2">
              <spacer name="horizontalSpacer_6">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="7" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelSpindleSpeedSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="maximum">
                <double>20000.000000000000000</double>
               </property>
               <property name="value">
                <double>15000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="bedLevelingDepthLabel">
               <property name="text">
                <string>Depth (in mm) :</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="bedLevelXYFeedRateLabel">
               <property name="text">
                <string>X/Y feed rate :</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelDepthSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>0.001000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.001000000000000</double>
               </property>
               <property name="value">
                <double>0.500000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="2">
              <spacer name="horizontalSpacer_10">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="6" column="0">
              <widget class="QLabel" name="bedLevelZFeedRate">
               <property name="text">
                <string>Z feed rate :</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>1.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100000.000000000000000</double>
               </property>
               <property name="value">
                <double>400.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="6" column="1">
              <widget class="QDoubleSpinBox" name="bedLevelZFeedRate_2">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>1.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100000.000000000000000</double>
               </property>
               <property name="value">
                <double>30.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="5" column="2">
              <spacer name="horizontalSpacer_11">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="6" column="2">
              <spacer name="horizontalSpacer_12">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="bedLevelFileToCreateLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>File to create :</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="bedLevelFileToCreateField"/>
             </item>
             <item>
              <widget class="QPushButton" name="bedLevelFileSelectButton">
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="bedLevelCreatePushButton">
            <property name="text">
             <string>Create...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="feedrateTweakerGroupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Feed Rate Tweaking and G-Code Cleanup</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="feedRateTweakingInputFileLabel">
               <property name="text">
                <string>Input file :</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="feedRateTweakingInputFileField"/>
             </item>
             <item>
              <widget class="QPushButton" name="feedRateTweakingInputFileButton">
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="feedRateTweakingCleanUpGcodeGroupCheckBox">
             <property name="title">
              <string>Clean up G-code</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QCheckBox" name="feedRateTweakingAllFeedRatesAlignedCheckBox">
                <property name="text">
                 <string>Feed Rates on the same line as the G-codes that use them</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="feedRateTweakingReplaceM05CheckBox">
                <property name="text">
                 <string>Replace M05 at the end of the file</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="feedRateTweakingRedefineFeedRateGroupCheckBox">
             <property name="title">
              <string>Redefine Feed Rates</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QCheckBox" name="feedRateTweakerOnlyReplaceFeedRateCheckBox">
                <property name="text">
                 <string>Only replace existing feed rate instances</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QGridLayout" name="gridLayout_2">
                <item row="0" column="0">
                 <widget class="QLabel" name="feedRateTweakingRedefineXYLabel">
                  <property name="text">
                   <string>X/Y feed rate :</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QDoubleSpinBox" name="feedRateTweakerXYFeedRateSpinBox">
                  <property name="decimals">
                   <number>4</number>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>400.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <spacer name="horizontalSpacer_7">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="feedRateTweakingZFeedRateLabel">
                  <property name="text">
                   <string>Z feed rate :</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QDoubleSpinBox" name="feedRateTweakingZFeedRateSpinBox">
                  <property name="decimals">
                   <number>4</number>
                  </property>
                  <property name="minimum">
                   <double>0.000100000000000</double>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>30.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <spacer name="horizontalSpacer_8">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="feedRateTweakerOutputFileLabel">
               <property name="text">
                <string>Output File :</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="feedRateTweakerOutputFileField"/>
             </item>
             <item>
              <widget class="QPushButton" name="feedRateTweakerOutputFileButton">
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <spacer name="horizontalSpacer_9">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="feedRateTweakingCreateButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Create...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>597</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuGcode">
    <property name="title">
     <string>&amp;G-Code</string>
    </property>
    <addaction name="actionCreate_Bed_Leveling_G_code"/>
    <addaction name="actionAdd_Edit_feed_rates_in_a_G_code_file"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <addaction name="menuGcode"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionCreate_Bed_Leveling_G_code">
   <property name="text">
    <string>&amp;Create Milling Bed Leveling G-code</string>
   </property>
  </action>
  <action name="actionAdd_Edit_feed_rates_in_a_G_code_file">
   <property name="text">
    <string>&amp;Add/Edit feed rates in a G-code file</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
